addons/l10n_in/models/res_partner.py

```diff
@@ -44,3 +44,13 @@ def onchange_vat(self):
     def _commercial_fields(self):
         res = super()._commercial_fields()
         return res + ['l10n_in_gst_treatment']
+
+    def check_vat_in(self, vat):
+        """
+            This GST number is used as test credentials for EDI
+            but this is not a valid number as per the regular expression
+            so we bypass this specific GST number.
+        """
+        if vat == "36AABCT1332L011":
```

Comment:
Do you need to use this in production? I see occurrences of `36AABCT1332L011` only in demo and test files, so if not needed, maybe you can make this change in the test file?

Reply:
@mart-e we need this when we give a demo to customer.

Reply:
no, we must need this number for the demo.
it's given by the provider.

Reply:
Maybe use a toplevel constant so that the name can clearly document the purpose, rather than have a less self-documenting docstring? e.g. `TEST_GST_NUMBER`. There can be a comment or docstring attached to that (possibly linking to an official documentation?), and separately the `check_vat_in` can have a smaller comment indicating that the test GST is considered always valid.

One thing which is slightly worrysome is what happens if this number gets used in non-demo mode, what would the impact be on the system? Would it just be useless because you can put it in the VAT/GST field but then none of the systems will care for it or can it have a larger impact?

Reply:
@xmo-odoo if anyone uses this number in production then the Government EDI server gives an error that this number is not valid.

Reply:
@xmo-odoo code update

Reply:
> if anyone uses this number in production then the Government EDI server gives an error that this number is not valid.

But they'll do that for any number which doesn't pass the normal check no?

Reply:
not for any number just for this one number only.

Reply:
@xmo-odoo so this okay for now?

Reply:
Better, but the semantics of this magic GST number still seem very confused / unclear, linking an official guidance / documentation would be really useful. I know that type of scheme exists (e.g. it's common with CC numbers for testing payment APIs), but usually there's a documentation we can link to for whatever dev follows.

Reply:
@xmo-odoo this number is shared by the provider and there is no documentation for this